brent/graph.py

```diff
@@ -45,9 +45,9 @@ def __init__(self, dataframe: pd.DataFrame):
         dag = DAG(df).add_edge("a", "b").add_edge("b", "c").add_edge("c","d")
         ```
         """
-        self.df = dataframe
+        self._df = dataframe
```

Comment:
yep. good one.

Reply:
come to think of it, why don't we do a count here? i cannot come up with a single good reason why we'd need every single row here.

Reply:
I also think we don't ever need every single row, but wasn't sure how `df` was being used throughout the other parts of the code. That's why I decoupled a bit first. 

does `Dag.from_observations(df)` sound like a good method from where to do the `group_by().count()`?

Reply:
sure.